mcs/class/System/Mono.Net.Security/SystemCertificateValidator.cs

```diff
@@ -45,7 +45,8 @@ static SystemCertificateValidator ()
 #elif MONODROID
 			is_macosx = false;
 #else
-			is_macosx = System.IO.File.Exists (OSX509Certificates.SecurityLibrary);
+			// make it work on wine too.
+			is_macosx = Environment.OSVersion.Platform == PlatformID.MacOSX;
```

Comment:
Under Wine, we use `mozroots` to create the certificate store under `<bottle>/drive_c/users/Public/Application Data/.mono/certs`.

Some time ago, I followed somehow the logic of the certificate validation process and I pinpointed why SSL didn't work on Wine. I implemented some reflection code to fix it. With this code in place the problem disappeared.

Find here an exerpt of the code I used to fix the bug:

```
			private static void					FixOsxWineCertificateValidation()
			{
				// There is a bug in mono-windows when running on OSX + Wine
				// The problem is that the discrimination of the platform is wrong on OSX + Wine.

				var macosxField = MonoUser.CertificateValidatorTypes
					.NonNull ()
					.Select (type => MonoUser.CertificateValidatorMacOsxField (type))
					.NonNull ()
					.FirstOrDefault ();

				if (macosxField != null)
				{
					RuntimeHelpers.RunClassConstructor (macosxField.DeclaringType.TypeHandle);
					var is_macosx = (bool) macosxField.GetValue (null);
					if (is_macosx)
					{
						// Force service point manager to use same SSL validation code as under mono-windows
						TraceMT.WriteLine ("#### Platform.FixOsxWineCertificateValidation -- done");
						macosxField.SetValue (null, false);
					}
				}
			}
			private static IEnumerable<Type>	CertificateValidatorTypes
			{
				get
				{
					yield return typeof (Uri).Assembly.GetType ("Mono.Net.Security.SystemCertificateValidator");
					yield return typeof (Uri).Assembly.GetType ("Mono.Net.Security.ChainValidationHelper");
					yield return typeof (ServicePointManager)
						.GetNestedTypes (BindingFlags.NonPublic)
						.Where (t => t.Name == "ChainValidationHelper")
						.SingleOrDefault ();
				}
			}
			private static FieldInfo			CertificateValidatorMacOsxField(Type type)
			{
				var field = type.GetField ("is_macosx", BindingFlags.NonPublic | BindingFlags.Static);
				TraceMT.WriteLine ($">>>> Platform.ChainValidationMacOsxFieldInfo({type.Name}) {(field == null ? "NOT " : "")} FOUND");
				return field;
			}

```